orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+            *  will be called */
+            opal_event_set(orte_event_base,
+                       &orte_rml_ofi.ofi_conduits[cur_conduit].progress_event,
+                       orte_rml_ofi.ofi_conduits[cur_conduit].fd,
+                       OPAL_EV_READ|OPAL_EV_PERSIST,
+                       cq_progress_handler,
+                       &orte_rml_ofi.ofi_conduits[cur_conduit]);
+            opal_event_add(&orte_rml_ofi.ofi_conduits[cur_conduit].progress_event, 0);
+            orte_rml_ofi.ofi_conduits[cur_conduit].progress_ev_active = true;
+           /** update the number of conduits in the ofi_conduits[] array **/
+           opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                 "%s:%d Conduit id - %d created ",__FILE__,__LINE__,orte_rml_ofi.conduit_open_num);
+           orte_rml_ofi.conduit_open_num++;
+        } 
+
+    }
+    /**
+     * Free providers info since it's not needed anymore.
+    */
+    fi_freeinfo(hints);
+    hints = NULL;
+    /* only if atleast one conduit was successfully opened then return the module */
+    if (0 < orte_rml_ofi.conduit_open_num ) {
+        opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                    "%s:%d conduits openened=%d returning orte_rml_ofi.super",
+                    __FILE__,__LINE__,orte_rml_ofi.conduit_open_num);
+
+        OBJ_CONSTRUCT(&orte_rml_ofi.exceptions, opal_list_t);
+        OBJ_CONSTRUCT(&orte_rml_ofi.recv_msg_queue_list,opal_list_t);
+        init_done = true;
+        return &orte_rml_ofi.super;
+    } else {
+        opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                    "%s:%d Failed to open any conduits",__FILE__,__LINE__);
+        return NULL;
+    }
+
+}
+
+
+int
+orte_rml_ofi_enable_comm(void)
+{
+    int ret;
+
+    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                        " %s - orte_rml_enable_comm() begin", 
+                        ORTE_NAME_PRINT(ORTE_PROC_MY_NAME));
+    /* enable the base receive to get updates on contact info */
+    orte_rml_base_comm_start();
+    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                        " %s - orte_rml_enable_comm() completed", 
+                        ORTE_NAME_PRINT(ORTE_PROC_MY_NAME));
+
+    return ORTE_SUCCESS;
+}
+
+
+void
+orte_rml_ofi_fini(void)
+{
+    opal_list_item_t *item;
+    uint8_t conduit_id;
+
+
+    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                        " %s -  orte_rml_ofi_fini() begin ", 
+                        ORTE_NAME_PRINT(ORTE_PROC_MY_NAME));
+
+    while (NULL !=
+           (item = opal_list_remove_first(&orte_rml_ofi.exceptions))) {
+        OBJ_RELEASE(item);
+    }
+    OBJ_DESTRUCT(&orte_rml_ofi.exceptions);
+    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                        " %s -  orte_rml_ofi_fini() completed ", 
+                        ORTE_NAME_PRINT(ORTE_PROC_MY_NAME));
```

Comment:
releasing all the conduit related allocations in rml_ofi_component_close() as they are created in rml_ofi_component_open()